### ISSUE TITLE
Convert to NPM module / change publishing procedure

### ISSUE THREAD

**Issue by alansemenov** [enonic/lib-admin-ui#2130]
When an application that uses lib-admin-ui is built in dev mode, Gradle first builds lib-admin-ui locally, then creates a JAR file, then triggers the build of the app itself, which unpacks lib-admin-ui's JAR file and uses its resources in the build.
This is inefficient because in this case we could simply use lib-admin-ui resources directly, without spending time to build, then unpack the JAR file.
lib-admin-ui should be changed to an ordinary NPM module that is published to the NPM registry every time we release a new version. A SNAPSHOT of lib-admin-ui should be deployed to an NPM registry in our Artifactory (we can then remove it from Maven repository).
In case of snapshots (when an application needs to use a SNAPSHOT of lib-admin-ui but doesn't have access to its sources locally) we 

So, to sum up.

**Changes in lib-admin-ui:**
1. lib-admin-ui should be made an NPM module (under `@enonic` namespace).
2. We should set up NPM registry in the Artifactory which will store the snapshots.
3. Publishing of lib-admin-ui should be changed to publish snapshots to the Artifactory's NPM registry and publish releases to the central NPM registry.

**Changes in apps that are using lib-admin-ui** (requires additional tasks):
1. The build should be changed to access lib-admin-ui resources directly if they exist locally. JAR should not be needed.
2. If an app is using a SNAPSHOT lib-admin-ui, it should be fetched from our Artifactory's NPM registry
3. Otherwise lib-admin-ui should be fetched as an ordinary NPM module from the central NPM registry.
